### PR TITLE
feature/tooltip > tooltip and renderToPortal from our old repo moved over

### PR DESCRIPTION
**Pull request by jeesunikim**
* Comments added in each file
* `renderToPortal` is from our old `stellar.org` repo
* There's a bug when the icon is hovered for the first time, the page is re-rendering.

### CODE REVIEW

File: src/components/CodeExample.js [hunk 1]
```diff
@@ -42,14 +44,11 @@ const OptionsContainer = styled.div`
...
   align-items: center;
-
-  & > * {
-    margin-left: 1rem;
-  }
 `;
 
 const MethodContentEl = styled.div`
   position: relative;
+  overflow: auto;
```
Comment (jeesunikim) [inline src/components/CodeExample.js]: Added to have `border-radius: 4px` visible

File: src/components/CodeExample.js [hunk 2]
```diff
@@ -130,28 +125,55 @@ const CodeSnippet = ({ codeSnippets, title, href }) => {
...
+  const [dimension, setDimension] = React.useState({ right: 0, top: 0 });
+
   const codeSnippetsArr = React.Children.toArray(codeSnippets);
   const selectedSnippet =
     codeSnippetsArr.find(
       (snippet) => snippet.props["data-language"] === activeLang,
     ) || codeSnippetsArr[0];
   const SelectedSnippetStr = mdxJsxToString(selectedSnippet);
 
+  const CopyToClipboardRef = React.useCallback((node) => {
```
Comment (jeesunikim) [inline src/components/CodeExample.js]: As React recommended, using `useCallback` to get the copy to clipboard icon's DOM position

File: src/components/CodeExample.js [hunk 3]
```diff
@@ -130,28 +125,55 @@ const CodeSnippet = ({ codeSnippets, title, href }) => {
...
     const langCookie = getCookie("lang");
+    let timer;
     if (langCookie) {
       setActiveLang(langCookie);
     } else {
       setActiveLang(availableLangs[0]);
     }
-  }, [availableLangs]);
+
+    if (isCopied) {
```
Comment (jeesunikim) [inline src/components/CodeExample.js]: Once `CopyToClipboard` icon gets clicked, `isCopied` is set to true then its value gets set to false in order to show its previous text `Click to Copy`

File: src/components/CodeExample.js [hunk 4]
```diff
@@ -192,6 +223,9 @@ const CodeSnippet = ({ codeSnippets, title, href }) => {
         <DocPrismStyles isCodeSnippet />
         {selectedSnippet}
       </ContentEl>
+      <Tooltip in={isHovered} isCopied={isCopied} parentDimension={dimension}>
```
Comment (jeesunikim) [inline src/components/CodeExample.js]: `CopyToClipboard` icon's dimension gets passed to Tooltip here

File: src/components/CodeExample.js [hunk 5]
```diff
@@ -42,14 +44,11 @@ const OptionsContainer = styled.div`
   display: flex;
   position: relative;
   align-items: center;
-
-  & > * {
-    margin-left: 1rem;
-  }
```
Comment (piyalbasu) [inline src/components/CodeExample.js]: 👍

**Pull request merged**